File: internal/cli/start.go
package cli

import (
	"bytes"
	"fmt"
	"io"
	"strings"
)

// StartResult is the JSON output shape for the start command.
type StartResult struct {
	OK        bool        `json:"ok"`
	Steps     []StartStep `json:"steps"`
	NextSteps []string    `json:"next_steps"`
}

// StartStep represents a single phase of the start flow.
type StartStep struct {
	Name   string `json:"name"`
	Status string `json:"status"`
	Note   string `json:"note,omitempty"`
}

func handleStart(args []string, stdout io.Writer, stderr io.Writer) int {
	root := "."
	profile := "minimal"
	apply := false
	skipDoctor := false
	skipExamples := false
	jsonMode := false
	wizardWithCard := ""

	for i := 0; i < len(args); i++ {
		switch args[i] {
		case "-h", "--help":
			fmt.Fprintln(stderr, "usage: xh start [--root <path>] [--profile minimal|standard|full|deep] [--apply] [--skip-doctor] [--skip-examples] [--json] [--wizard-with-card <task_id>]")
			return ExitUsage
		case "--root":
			if i+1 < len(args) {
				root = args[i+1]
				i++
			}
		case "--profile":
			if i+1 < len(args) {
				profile = args[i+1]
				i++
			}
		case "--apply":
			apply = true
		case "--skip-doctor":
			skipDoctor = true
		case "--skip-examples":
			skipExamples = true
		case "--json":
			jsonMode = true
		case "--wizard-with-card":
			if i+1 < len(args) {
				wizardWithCard = args[i+1]
				i++
			}
		}
	}

	validProfiles := map[string]bool{"minimal": true, "standard": true, "full": true, "deep": true}
	if !validProfiles[profile] {
		fmt.Fprintf(stderr, "usage: xh start [--root <path>] [--profile minimal|standard|full|deep] [--apply] [--skip-doctor] [--skip-examples] [--json] [--wizard-with-card <task_id>]\n")
		fmt.Fprintf(stderr, "invalid profile: %s\n", profile)
		return ExitUsage
	}

	initProfile := profile
	if profile == "deep" {
		initProfile = "full"
	}

	var steps []StartStep

	// Step 1: Doctor
	if !skipDoctor {
		var doctorStdout, doctorStderr bytes.Buffer
		doctorCode := handleDoctor([]string{"--root", root, "--json"}, &doctorStdout, &doctorStderr)
		status := "passed"
		if doctorCode != ExitOK {
			status = "failed"
		}
		note := "workspace healthy"
		if doctorCode != ExitOK {
			note = "workspace has issues"
		}
		if doctorStderr.Len() > 0 {
			note = strings.TrimSpace(doctorStderr.String())
		}
		steps = append(steps, StartStep{Name: "doctor", Status: status, Note: note})
	}

	// Step 2: Examples verify
	if !skipExamples {
		var exStdout, exStderr bytes.Buffer
		exCode := handleExamplesVerify([]string{"--json"}, &exStdout, &exStderr)
		status := "passed"
		if exCode != ExitOK {
			status = "failed"
		}
		note := "all golden examples passed"
		if exCode != ExitOK {
			note = "some golden examples failed"
			if exStderr.Len() > 0 {
				note = strings.TrimSpace(exStderr.String())
			}
		}
		steps = append(steps, StartStep{Name: "examples_verify", Status: status, Note: note})
	}

	// Step 3: Init wizard
	{
		initArgs := []string{root, "--wizard", "--wizard-profile", initProfile}
		if !apply {
			initArgs = append(initArgs, "--wizard-dry-run")
		}
		if wizardWithCard != "" {
			initArgs = append(initArgs, "--wizard-with-card", wizardWithCard)
		}
		var initStdout, initStderr bytes.Buffer
		initCode := handleInit(initArgs, &initStdout, &initStderr)
		status := "passed"
		if initCode != ExitOK {
			status = "failed"
		}
		note := "init wizard previewed"
		if apply {
			note = "init wizard applied"
		}
		if initCode != ExitOK {
			note = "init wizard failed"
			if initStderr.Len() > 0 {
				note = strings.TrimSpace(initStderr.String())
			}
		}
		steps = append(steps, StartStep{Name: "init_wizard", Status: status, Note: note})
	}

	// Determine overall ok
	ok := true
	for _, s := range steps {
		if s.Status != "passed" && s.Status != "skipped" {
			ok = false
			break
		}
	}

	nextSteps := []string{
		"Run your first verification: xh check --card completion-card.yaml",
		"Read the docs: docs/GETTING_STARTED.md",
	}

	if jsonMode {
		result := StartResult{
			OK:        ok,
			Steps:     steps,
			NextSteps: nextSteps,
		}
		if err := WriteJSON(stdout, result); err != nil {
			return ExitError
		}
	} else {
		WriteLine(stdout, "# xh start - Guided onboarding")
		WriteLine(stdout, "")
		stepNum := 1
		if !skipDoctor {
			WriteLine(stdout, "Step %d/4: doctor", stepNum)
			WriteLine(stdout, "  status: %s", steps[stepNum-1].Status)
			if steps[stepNum-1].Note != "" {
				WriteLine(stdout, "  note: %s", steps[stepNum-1].Note)
			}
			WriteLine(stdout, "")
			stepNum++
		}
		if !skipExamples {
			WriteLine(stdout, "Step %d/4: examples verify", stepNum)
			WriteLine(stdout, "  status: %s", steps[stepNum-1].Status)
			if steps[stepNum-1].Note != "" {
				WriteLine(stdout, "  note: %s", steps[stepNum-1].Note)
			}
			WriteLine(stdout, "")
			stepNum++
		}
		WriteLine(stdout, "Step %d/4: init wizard", stepNum)
		// init wizard is always the last step in steps slice
		last := steps[len(steps)-1]
		WriteLine(stdout, "  status: %s", last.Status)
		if last.Note != "" {
			WriteLine(stdout, "  note: %s", last.Note)
		}
		WriteLine(stdout, "")
		WriteLine(stdout, "Next steps:")
		for _, s := range nextSteps {
			WriteLine(stdout, "  - %s", s)
		}
	}

	if ok {
		return ExitOK
	}
	return ExitError
}

File: packages/cli/src/commands/start.ts
import { Command } from "commander";
import * as path from "node:path";
import {
  type Lang,
  resolveLang,
  startTitle,
  startStepLabel,
  startNextStepsTitle,
  startFirstVerification,
  startReadDocs,
} from "../i18n.js";

interface StartStep {
  name: string;
  status: string;
  note?: string;
}

interface StartResult {
  ok: boolean;
  steps: StartStep[];
  next_steps: string[];
}

/**
 * TS start is a preview/planner: it lists planned steps without executing them.
 * The Go CLI executes doctor, examples verify, and init wizard directly.
 */
export function startCommand(): Command {
  return new Command("start")
    .description(
      "Guided onboarding: doctor, examples verify, init wizard, next steps"
    )
    .option("--root <path>", "Repository root", process.cwd())
    .option(
      "--profile <profile>",
      "Install profile (minimal, standard, deep)",
      "minimal"
    )
    .option("--apply", "Apply changes (init wizard without dry-run)", false)
    .option("--skip-doctor", "Skip doctor step", false)
    .option("--skip-examples", "Skip examples verify step", false)
    .option("--json", "Output JSON instead of text", false)
    .option(
      "--wizard-with-card <task_id>",
      "Scaffold a completion card on apply"
    )
    .option("--lang <code>", "Language", "en")
    .action(
      async (
        opts: {
          root: string;
          profile: string;
          apply: boolean;
          skipDoctor: boolean;
          skipExamples: boolean;
          json: boolean;
          wizardWithCard?: string;
          lang?: string;
        },
        cmd: Command
      ) => {
        const validProfiles = ["minimal", "standard", "full", "deep"];
        if (!validProfiles.includes(opts.profile)) {
          console.error(
            "usage: xh start [--root <path>] [--profile minimal|standard|full|deep] [--apply] [--skip-doctor] [--skip-examples] [--json] [--wizard-with-card <task_id>]"
          );
          console.error(`invalid profile: ${opts.profile}`);
          process.exit(2);
        }

        const lang: Lang = resolveLang(opts, cmd.parent?.opts() ?? {});

        const steps: StartStep[] = [];

        // Step 1: Doctor
        if (!opts.skipDoctor) {
          steps.push({
            name: "doctor",
            status: "planned",
            note: `xh doctor --root ${path.resolve(opts.root)} --json`,
          });
        }

        // Step 2: Examples verify
        if (!opts.skipExamples) {
          steps.push({
            name: "examples_verify",
            status: "planned",
            note: "xh examples verify --json",
          });
        }

        // Step 3: Init wizard
        const mode = opts.apply ? "apply" : "preview";
        const initNote = `xh init ${path.resolve(opts.root)} --wizard --wizard-profile ${opts.profile}${
          opts.apply ? "" : " --wizard-dry-run"
        }${opts.wizardWithCard ? ` --wizard-with-card ${opts.wizardWithCard}` : ""}`;
        steps.push({
          name: "init_wizard",
          status: "planned",
          note: initNote,
        });

        const nextStepsText = [
          startFirstVerification(lang),
          startReadDocs(lang),
        ];
        const nextStepsJSON = [
          "Run your first verification: xh check --card completion-card.yaml",
          "Read the docs: docs/GETTING_STARTED.md",
        ];

        if (opts.json) {
          const result: StartResult = {
            ok: true,
            steps,
            next_steps: nextStepsJSON,
          };
          console.log(JSON.stringify(result, null, 2));
        } else {
          console.log(`# ${startTitle(lang)}`);
          console.log("");
          if (!opts.skipDoctor) {
            console.log(`Step 1/4: ${startStepLabel("doctor", lang)}`);
            console.log(`  -> ${steps[0].note}`);
            console.log("");
          }
          if (!opts.skipExamples) {
            const idx = opts.skipDoctor ? 0 : 1;
            console.log(
              `Step ${idx + 1}/4: ${startStepLabel("examples_verify", lang)}`
            );
            console.log(`  -> ${steps[idx].note}`);
            console.log("");
          }
          const initIdx = steps.findIndex((s) => s.name === "init_wizard");
          console.log(
            `Step ${initIdx + 1}/4: ${startStepLabel("init_wizard", lang)} (${mode})`
          );
          console.log(`  -> ${steps[initIdx].note}`);
          console.log("");
          console.log(startNextStepsTitle(lang));
          for (const s of nextStepsText) {
            console.log(`  - ${s}`);
          }
        }
      }
    );
}
